models/template_helpers.go

```diff
@@ -183,6 +183,10 @@ func RemoveTemplate(q *reform.Querier, name string) error {
 		return err
 	}
 
+	if _, err := q.FindOneFrom(RuleTable, "template_name", name); err == nil {
+		return errors.Errorf("failed to delete rule template, as it is being used by a rule")
+	}
```

Comment:
Error message is true only wher returned error is `reform.ErrNoRows`, otherwise reason can be different.

Reply:
but its only checking for when we don't have an error `err == nil` (meaning that there was a rule with the template name), in case there is an error that means there are no rules with that template name and we can carry on deleting the template instead of returning that  error.

Reply:
🤦🤦🤦 You right

Reply:
We will also delete the template if there was some temporary error like lost database connection, etc. Artem is right – we should check only for reform.ErrNoRows

Reply:
I think the current code can be confusing should I do something like this would this be more readable?

```go

func DeleteTemplate() {
if templateInUse(name) {
   return errors.Errorf("failed to delete rule template, as it is being used by a rule")
}
}


func templateInUse(name string) bool {
  if _, err := q.FindOneFrom(RuleTable, "template_name", name); err != nil {
		return false
   }
   return true
}
```

Reply:
ignore the comment, I posted it without refreshing the page

Reply:
added error check https://github.com/percona/pmm-managed/pull/599/commits/a91f584ae772c6ee4d6218302c11ddaf6d6aafe5